tests/Behat/PoetryTranslationContext.php

```diff
@@ -0,0 +1,256 @@
+    $languages = $this->getLanguagesFromNames($languages);
+    if (!$languages) {
+      throw new \Exception('The specified languages cannot be found');
+    }
+
+    $langcodes = array_keys($languages);
+    foreach ($langcodes as $langcode) {
+      $this->getSession()->getPage()->checkField("languages[$langcode]");
+    }
+  }
+
+  /**
+   * Checks that the Poetry jobs for a given node got created for the languages.
+   *
+   * @param string $title
+   *   The node title.
+   * @param string $languages
+   *   The languages.
+   *
+   * @Then the Poetry request jobs to translate :title should get created for :languages
+   */
+  public function thePoetryRequestJobsShouldGetCreated(string $title, string $languages): void {
+    $languages = $this->getLanguagesFromNames($languages);
+    $node = $this->getNodeByTitle($title);
+    $query = $this->getEntityJobsQuery($node);
+    $query->isNull('job.poetry_state');
+    $result = $query->execute()->fetchAllAssoc('target_language');
+    if (count(array_intersect_key($result, $languages)) !== count($languages)) {
+      throw new \Exception('The jobs have not been created for these languages.');
+    }
+  }
+
+  /**
+   * Accepts the translation jobs.
+   *
+   * This mimics a notification that has come from the Poetry service which
+   * contains an acceptance status notification.
+   *
+   * @param string $title
+   *   The node title.
+   * @param string $languages
+   *   The languages.
+   *
+   * @Given the Poetry translation request of :title in :languages gets accepted
+   */
+  public function thePoetryTranslationRequestGetAccepted(string $title, string $languages): void {
+    $languages = $this->getLanguagesFromNames($languages);
+    $node = $this->getNodeByTitle($title);
+    $query = $this->getEntityJobsQuery($node);
+    $query->condition('job.target_language', array_keys($languages), 'IN');
+    $query->isNull('job.poetry_state');
+    $result = $query->execute()->fetchAllAssoc('tjid');
+    $jobs = Job::loadMultiple(array_keys($result));
+    /** @var \Drupal\tmgmt\JobInterface $job */
+    $job = reset($jobs);
+
+    $accepted = [];
+    foreach ($languages as $language) {
+      $accepted[] = [
+        'code' => strtoupper($language->getId()),
+        'date' => '30/08/2050 23:59',
+        'accepted_date' => '30/09/2050 23:59',
+      ];
+    }
+    $status_notification = \Drupal::service('oe_translation_poetry_mock.fixture_generator')->statusNotification($job->get('poetry_request_id')->first()->getValue(), 'ONG', $accepted);
+
+    $this->performNotification($status_notification);
+  }
+
+  /**
+   * Receives the translations.
+   *
+   * This mimics a notification that has come from the Poetry service which
+   * contains the translated content.
+   *
+   * @param string $title
+   *   The node title.
+   * @param string $languages
+   *   The languages.
+   *
+   * @Given the Poetry translation(s) of :title in :languages get sent by Poetry
```

Comment:
The step wording is a bit weird since on our end there is not sending, I think I would reword it to be something like "he Poetry translation(s) of :title in :languages are received). (Nitpicking here, feel free to ignore.)